app/forms/totp_setup_form.rb

```diff
@@ -29,8 +29,12 @@ def submit
   def valid_totp_code?
     # The two_factor_authentication gem raises an error if the secret is nil.
     return false if secret.nil?
-    new_timestamp = Db::AuthAppConfiguration::Confirm.call(secret, code)
-    create_auth_app(user, secret, new_timestamp, name) if new_timestamp
+    new_timestamp = Db::AuthAppConfiguration.confirm(secret, code)
+    if new_timestamp
+      create_auth_app(user, secret, new_timestamp, name) if new_timestamp
```

Comment:
this trailing `if` is redundant because now its inside a block guarding the same thing

```suggestion
      create_auth_app(user, secret, new_timestamp, name)
```